src/Common/src/Interop/Unix/System.Private.CoreLib.Native/Interop.MemAllocFree.cs

```diff
@@ -8,10 +8,12 @@
 
 internal static partial class Interop
 {
+    internal const int HEAP_ZERO_MEMORY = 0x8;
+    
     internal unsafe partial class Sys
     {
         [DllImport(Interop.Libraries.CoreLibNative, EntryPoint = "CoreLibNative_MemAlloc")]
-        internal static extern IntPtr MemAlloc(UIntPtr sizeInBytes);
+        internal static extern IntPtr MemAlloc(UIntPtr sizeInBytes, bool zeroInitialize);
```

Comment:
The zero initialization should be a new method, not a flag on raw memalloc - so that only the few places that actually need it pay for it.